theme/static/js/resourceLandingAjax.js

```diff
@@ -842,8 +842,9 @@ function unzip_irods_file_ajax_submit(res_id, zip_with_rel_path) {
             zip_with_rel_path: zip_with_rel_path,
             remove_original_zip: "false"
         },
-        success: function (result) {
-            // TODO: handle "File already exists" errors
+        success: function (task) {
+            notificationsApp.registerTask(task);
+            notificationsApp.show();
```

Comment:
I am wondering where you handle the file browser refresh when the task succeeds.

Reply:
I raised questions about whether we should even be handling a file browser refresh given the new async framework.  I'd be irritated if a long running async task completed, while I was reading or editing a resource, and refreshed my browser.

Reply:
@sblack-usu Your point is definitely a valid one, but on the other side of the argument, for a short running async task, it might be irritating as well that the whole page has to be refreshed by manually clicking a button each time to see the changes. I don't have strong opinions on one or the other, but it appears easier and more maintainable to provide a refresh button for users to manually refresh the whole page to see the changes. We do have to update the notification app to add such a page refresh button, right?

Reply:
I mentioned @cuahsimarko on your review to discuss this issue.  We need a frontend developer to think more about the UX for this.  Also, there may be some functionality in vue that we could harness that you and I are not aware of.

Reply:
I discussed this in detail with Hong last week and HydroShare is not ready to implement modern practices in this area. I have a future project to extend best practices being implemented in new Discover only across all of HydroShare and make the frameworks and developer practices available to any section of HydroShare code.

Reply:
I understand we can't use all of new modern practices across hydroshare yet, that was not what I was asking.  We need your UX design input here before we get the updated tools in all of hydroshare.

Reply:
I discussed this with Hong and Mauriel's initial sentiment and implementation is sound. Going any further without extending the modern practices to the underlying foundation of HS isn't practical. Hong has achieved a good stop-gap here.

Reply:
so, you think an async task should do things like refresh the page when it completes?  Even though, the current system currently doesn't know which page the user is on when it initiates a refresh?

Reply:
@sblack-usu That is the problem. It is not only refreshing the page. It can also be redirecting to another page, e.g., for resource deletion, or some other actions. Once an action/task is triggered from another page and the task is registered with the notification app, the notification app does not have context in terms of what needs to be accomplished after the task is done. The download task is handled within the notification app, however, I don't think we should let the notification app to handle all actions after tasks are done, so I did it another way by having the task trigger app keeping the context and checking the task status from the notification app, then do what it needs to do after the task is done.